operator/flags.go

```diff
@@ -138,9 +115,74 @@ func init() {
 	flags.Bool(operatorOption.EnableMetrics, false, "Enable Prometheus metrics")
 	option.BindEnv(operatorOption.EnableMetrics)
 
-	flags.String(option.IPAM, ipamOption.IPAMHostScopeLegacy, "Backend to use for IPAM")
+	var defaultIPAM string
+	switch binaryName {
+	case "cilium-operator":
+		defaultIPAM = ipamOption.IPAMHostScopeLegacy
+	case "cilium-operator-aws":
+		defaultIPAM = ipamOption.IPAMENI
+	case "cilium-operator-azure":
+		defaultIPAM = ipamOption.IPAMAzure
+	case "cilium-operator-generic":
+		defaultIPAM = ipamOption.IPAMOperator
+	}
+
+	flags.String(option.IPAM, defaultIPAM, "Backend to use for IPAM")
 	option.BindEnv(option.IPAM)
 
+	rootCmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		ipamFlag := cmd.Flag(option.IPAM)
+		if !ipamFlag.Changed {
+			return nil
+		}
+		ipamFlagValue := ipamFlag.Value.String()
+
```

Comment:
@aanm I'm not sure how much sense this logic makes since binaries are in different images now, I'm tempted to just get rid of this and go for simple error messages... wdyt?

Reply:
Per slack discussion this is still useful, as image names actually match binary names.